pyccel/ast/cupyext.py

```diff
@@ -58,14 +59,29 @@
 )
 
 #==============================================================================
-class CupyArray(CudaNewArray):
+class CupyNewArray(CudaNewArray):
+    """ Class from which all Cupy functions which imply a call to Allocate
+    inherit
+    """
+    __slots__ = ('_arg','_dtype','_precision','_shape','_rank','_order', '_memory_location')
```

Comment:
I don't think this class should define all these slots. Why does a `NewArray` function only have 1 argument? Why can't any of these be class objects in sub classes?

Reply:
Fixed in [07b3260](https://github.com/pyccel/pyccel/pull/1300/commits/07b3260ba94aa741a57d5dee713ad0fd1b7b3039).